File: Conversion.cs
using System;

class Conversion
{
    private readonly Utils _utils;

    public Conversion(Utils utils)
    {
        _utils = utils;
    }

    public void ConvertCurrency()
    {
        Console.Write("Entrez le montant à convertir : ");
        double amount;
        while (!double.TryParse(Console.ReadLine(), out amount))
        {
            Console.WriteLine("Veuillez entrer un montant valide.");
        }

        Console.Write("Entrez la devise actuelle (ar/frc) : ");
        string currentCurrency = Console.ReadLine();

        Console.Write("Entrez la devise cible (ar/frc) : ");
        string targetCurrency = Console.ReadLine();

        // Logique pour convertir la devise
        double conversionRate = _utils.ConversionRate(currentCurrency, targetCurrency);
        double convertedAmount = amount * conversionRate;

        // Enregistrer la transaction de conversion
        string conversionTransaction = $"Date={_utils.GetCurrentDate()};Montant={amount} {currentCurrency} -> {convertedAmount} {targetCurrency}\n";
        System.IO.File.AppendAllText(_utils.GetTransactionFilePath(), conversionTransaction);

        Console.WriteLine($"Montant converti : {convertedAmount} {targetCurrency}");
    }
}

File: conversion.js
const readlineSync = require('readline-sync');
const fs = require('fs');
const utils = require('./utils');

function convertCurrency() {
    const amount = parseFloat(readlineSync.question('Entrez le montant à convertir: '));
    const currentCurrency = readlineSync.question('Entrez la devise actuelle (ar/frc): ');
    const targetCurrency = readlineSync.question('Entrez la devise cible (ar/frc): ');

    // Logique pour convertir la devise
    const conversionRate = utils.conversionRate(currentCurrency, targetCurrency);
    const convertedAmount = amount * conversionRate;

    // Enregistrer la transaction de conversion
    const conversionTransaction = `Date=${utils.getCurrentDate()};Montant=${amount} ${currentCurrency} -> ${convertedAmount} ${targetCurrency}\n`;
    fs.appendFileSync(utils.getTransactionFilePath(), conversionTransaction);

    console.log(`Montant converti: ${convertedAmount} ${targetCurrency}`);
}
module.exports = {
    convertCurrency
};